node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/Status.java

```diff
@@ -38,7 +41,8 @@ public Status(Generation generation,
         Objects.requireNonNull(hardwareDivergence, "Hardware divergence must be non-null");
         hardwareDivergence.ifPresent(s -> requireNonEmptyString(s, "Hardware divergence must be non-empty"));
         this.reboot = Objects.requireNonNull(generation, "Generation must be non-null");
-        this.vespaVersion = Objects.requireNonNull(vespaVersion, "Vespa version must be non-null");
+        this.vespaVersion = Objects.requireNonNull(vespaVersion, "Vespa version must be non-null").filter(v -> !Version.emptyVersion.equals(v));
```

Comment:
Maybe a strange place to have this, but the alternative is to have this check everywhere status is created (zk deserialization and NodePatcher at least)